control/wsrelay/client.go

```diff
@@ -0,0 +1,81 @@
+package wsrelay
+
+import (
+	"crypto/tls"
+	"net/url"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+)
+
+// Client is a websocket client
+type Client struct {
+	conn *websocket.Conn
+}
+
+// NewClient creates a new websocket client that can be interrupted
+// via SIGINT
+func NewClient(brokerAddr, path, secret string, useTLS bool, insecure bool) (*Client, error) {
+	// determine the scheme
+	var scheme string
+	if useTLS {
+		scheme = "wss"
+	} else {
+		scheme = "ws"
+	}
+
+	// construct the URL to connect to
+	u := url.URL{
+		Scheme: scheme,
+		Host:   brokerAddr,
+		Path:   path,
+	}
+
+	// set the secret in the query params
```

Comment:
Where does this drop that secret?

putting it into the URL parameters makes it visible to logs and various intermediates.